entry.go

```diff
@@ -29,6 +29,10 @@ type Entry struct {
 
 	// Message passed to Debug, Info, Warn, Error, Fatal or Panic
 	Message string
+
+	// A receive-only notification channel for knowing when all hooks have
+	// finished firing.
+	HooksDone <-chan struct{}
```

Comment:
I think a `sync.WaitGroup` would be more appropriate here. Also can you explain the purpose of exposing this? It's not clear to me what this is supposed to be used for. Hooks don't have an ID, so you could have many hooks being fired, one of them yours, and you'd be notified that one of those many hooks finished?